pkgs/os-specific/linux/intel-sgx/default.nix

```diff
@@ -0,0 +1,296 @@
+{ type, debug ? false }:
+assert builtins.elem type [ "sdk" "psw" ];
+
+{ lib
+, pkgs
+, cmake
+, stdenv
+, fetchurl
+, fetchFromGitHub
+, coreutils
+, autoconf
+, automake
+, libtool
+, curl
+, ocaml
+, ocamlPackages
+, which
+, file
+, git
+, getconf
+, gnum4
+, openssl
+, perl
+, protobuf
+, texinfo
+, bison
+, flex
+}:
+let
+  sgx = rec {
+    version = "2.13";
+    prebuiltUrl = "https://download.01.org/intel-sgx/sgx-linux/${version}";
+  };
+
+  dcap = rec {
+    version = "1.10";
+    prebuiltUrl = "https://download.01.org/intel-sgx/sgx-dcap/${version}/linux";
+  };
+
+  prebuilt = {
+    optlib = fetchurl {
+      url = "${sgx.prebuiltUrl}/optimized_libs_${sgx.version}.tar.gz";
+      sha256 = "sha256-ok/UKBR6//uGAww0dD6MuVMvfUhH5E6F/Z5DAx1PA1k=";
+    };
+
+    binutils = fetchurl {
+      url = "${sgx.prebuiltUrl}/as.ld.objdump.gold.r3.tar.gz";
+      sha256 = "sha256-eUljypD7BWHK8+0r7h2bo5QibzVWic3aKBYebgYgpxM=";
+    };
+
+    ae = fetchurl {
+      url = "${sgx.prebuiltUrl}/prebuilt_ae_${sgx.version}.tar.gz";
+      sha256 = "sha256-ggxxP2tFMbe6R1/bKI1hZkcnfVqxYBmYxtxQ12G4hp4=";
+    };
+
+    dcap = fetchurl {
+      url = "${dcap.prebuiltUrl}/prebuilt_dcap_${dcap.version}.tar.gz";
+      sha256 = "sha256-lT5oEKWLyV987ifFyC3QfwP0POLNInUAwsPyPZHaWgA=";
+    };
+  };
+in
+stdenv.mkDerivation {
+  inherit (sgx) version;
+  pname = "intel-sgx-${type}";
+
+  src = fetchFromGitHub {
+    owner = "intel";
+    repo = "linux-sgx";
+    rev = "sgx_${sgx.version}";
+    sha256 = "0iifafkh2hw2vy6ch50m5p0gk4nnmkppsc54ss61bhw09w6cfj1l";
+    fetchSubmodules = true;
+  };
+
+  postUnpack = ''
+    tar -zxf ${prebuilt.optlib} -C $sourceRoot
+    tar -zxf ${prebuilt.binutils} -C $sourceRoot
+    tar -zxf ${prebuilt.ae} -C $sourceRoot
+    tar -zxf ${prebuilt.dcap} -C $sourceRoot/external/dcap_source/QuoteGeneration/
+  '';
+
+  postPatch = ''
+    SGX_VERSION=$(awk '/STRFILEVER/ {print $3}' common/inc/internal/se_version.h|sed 's/^\"\(.*\)\"$/\1/')
+
+    substituteInPlace buildenv.mk \
+        --replace /bin/cp ${coreutils}/bin/cp
+    substituteInPlace external/dcap_source/QuoteGeneration/buildenv.mk \
+        --replace /bin/cp ${coreutils}/bin/cp
+
+    substituteInPlace external/openmp/Makefile \
+        --replace '$(RM) -rf openmp_code/final/build $(BUILD_DIR)/libsgx_omp.a' 'rm -f $(BUILD_DIR)/libsgx_omp.a' \
+        --replace '$(RM) -rf openmp_code' 'echo >/dev/null' \
+        --replace 'git clone' 'true || git clone'
+
+    substituteInPlace psw/ae/aesm_service/source/CMakeLists.txt \
+        --replace '/usr/bin/getconf' '${getconf}/bin/getconf'
+    
+    for i in linux/installer/common/sdk/pkgconfig/template/*.pc; do
+      substituteInPlace $i \
+        --replace /opt/intel/sgxsdk $out/usr/share/sgxsdk \
+        --replace @LIB_FOLDER_NAME@ lib64 \
+        --replace @SGX_VERSION@ $SGX_VERSION
+    done
+  '';
+
+  installPhase =
+    let build = "build/linux";
+    in
+    if type == "sdk" then ''
```

Comment:
Personally, I'd prefer a common base derivation with two distinct derivations over a flag. In fact, the SDK is a dependency of the PSW package and also in general they have different dependencies.